src/clj_jgit/porcelain.clj

```diff
@@ -461,6 +461,12 @@
      (-> repo .reset
          (.setRef ref)
          (.setMode ^ResetCommand$ResetType (reset-modes mode-sym))
+         (.call)))
+  ([^Git repo ref mode-sym path]
+     (-> repo .reset
+         (.setRef ref)
+         (.setMode ^ResetCommand$ResetType (reset-modes mode-sym))
```

Comment:
Mixing `modes` and `paths` is not allowed, so this should be removed along with the `mode-sym` arg. Which then leaves us with 2 overloads with the same arity. This is currently a bit of general problem with the library, probably time to tackle this finally, I'll open a separate issue for this for further discussion.